adagrams/game.py

```diff
@@ -1,13 +1,197 @@
+    '''
+    make_dict = {}
+    for character in array:
+        if character not in make_dict:
+            make_dict[character] = 1
+        else:
+            make_dict[character] += 1
+    return make_dict
+
+
 def draw_letters():
-    pass
+    # Convert letter pool dictionary to a list with repeated letters 
+    # the list above has 98 letters.
+    # make a random selection out of 98 letters of pool list.
+    # add the selected letter to the hand and add a counter that shows frequency of the letter in the hand
+    # draw hand should be a dict array with k/v as letters / freq
+    # check if freq of the selected letter is less than the freq of the pool
+    # repeat this 10 times!
+    # return a list of hand letters!
+    
+    letter_pool_list = []
+    hand_dict = {}
+    hand_list = []
+
+    for letter, frequency in LETTER_POOL.items():
+        for i in range(frequency):
+            letter_pool_list.append(letter)
+
+    # for i in range(10): Can't use for loop in indetermined number of loops
+    hand_counter = 1
+    while hand_counter <= 10:
+        draw = randint(0, 97)
+        letter = letter_pool_list[draw]
+
+        if letter not in hand_dict:
+            hand_dict[letter] = 1
+        else:
+            if hand_dict[letter] < LETTER_POOL[letter]:
+                hand_dict[letter] += 1
+            else:
+                continue
+        
+        hand_counter += 1
+
+    for letter, frequency in hand_dict.items():
+        for i in range(frequency):
+            hand_list.append(letter)
+
+    return hand_list
+
 
 def uses_available_letters(word, letter_bank):
-    pass
+    # first need to understand what are letters in word and how many of the letter do we have.
+    # then I need to compare each letter with my draw bank:
+    # if the letter in the word doesn't exist in the draw return false
+    # if the letter is in the draw bank then check if the frequency in the word 
+    # is more than the frequency of same letter in the draw bank
+    
+    word = word.upper()
+    word_dict = make_dict(word)
+    letter_bank_dict = make_dict(letter_bank)
+
+    for character, frequency in word_dict.items():
+        if  character not in letter_bank_dict:
+            return False
+        else:
+            if frequency > letter_bank_dict[character]:
+                return False
+    return True
+    
 
 def score_word(word):
-    pass
+    # compare the letters in the word with the letters in score chart. 
+    # add corresponding value to the total score
+    # Chech if the length of word is more than 7 
+    # (should we check if it is more than 10?)
+    # return the total score
+
+
+    # letter_bank = draw_letters() 
+    word = word.upper()
+    word_dict = make_dict(word)
+
+    # if uses_available_letters(word, letter_bank): # No need to check!
+    total_score = 0
+
+    for letter, frequency in word_dict.items():
+        total_score += score_chart[letter] * frequency
+
+    if len(word) >= 7:
```

Comment:
In the project docs, 8 points are added if the word is between 7 and 10 letters inclusive! Right now you'll add 8 points for 11+ letters as well! Is there a way that you could add this constraint to the conditional you have?